src/System.Private.ServiceModel/tests/Common/Infrastructure/xunit/IssueAttribute.cs

```diff
@@ -73,5 +86,71 @@ public override string GetSkipReason(ITestMethod testMethod)
             // Null means "don't skip"
             return null;
         }
+
+        // Returns 'true' if all tests with issues should be included (i.e. executed instead of skipped).
+        // If it returns 'false', s_includeAllTestsWithIssuesHash will be null or contain
+        // only the issue numbers to include
+        private static bool DetectIssuesToInclude()
+        {
+            string includeTestsWithIssues = TestProperties.GetProperty(TestProperties.IncludeTestsWithIssues_PropertyName);
+
+            // Empty string means don't include any tests with issues.
+            // In other words, all tests with [Issue] will be skipped.
+            if (String.IsNullOrEmpty(includeTestsWithIssues))
+            {
+                return false;
+            }
+
+            // The string "all" means include all tests with issues.
+            // In other words, all tests with [Issue] will be run and not skipped.
+            if (String.Equals(includeTestsWithIssues, "all", StringComparison.OrdinalIgnoreCase))
+            {
+                return true;
+            }
+
+            // Anything else is interpreted as a semicolon-separate list of
+            // issue numbers to include (i.e. not skip).
+            string[] issues = includeTestsWithIssues.Split(';');
+            foreach (string issue in issues)
+            {
+                if (String.IsNullOrEmpty(issue))
+                {
+                    continue;
+                }
+
+                int issueAsInt = 0;
+                if (!int.TryParse(issue, out issueAsInt))
```

Comment:
NIT: this check will allow -ve numbers through, but i don't think it does any harm even if negative numbers are passed